### PR TITLE
[Feature] Redis embedding store spring boot starter

### PR DESCRIPTION
**Pull request by Martin7-1**
Hi, I add `langchian4j-redis-spring-boot-starter` implementation. Test run correct in my local environment. 

btw, I'll add spring boot starters to the other vector stores in a few days. Waiting for your code reviews!

### CODE REVIEW

File: langchain4j-redis-spring-boot-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>dev.langchain4j</groupId>
+        <artifactId>langchain4j-spring</artifactId>
+        <version>0.26.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>langchain4j-oredis-spring-boot-starter</artifactId>
```
Comment (langchain4j) [inline langchain4j-redis-spring-boot-starter/pom.xml:13]:
```suggestion
    <artifactId>langchain4j-redis-spring-boot-starter</artifactId>
```

File: langchain4j-redis-spring-boot-starter/src/main/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreProperties.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    static final String PREFIX = "langchain4j.redis";
+
+    private String host;
+    private Integer port;
+    private String user;
+    private String password;
+    private String indexName;
+    private Integer dimension;
+    private List<String> metadataFieldsName = new ArrayList<>();
```
Comment (langchain4j) [inline langchain4j-redis-spring-boot-starter/src/main/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreProperties.java:23]:
```suggestion
    private List<String> metadataKeys = new ArrayList<>();
```

File: langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        redis.stop();
+    }
+
+    @Test
+    void should_provide_redis_vector_store() {
+        TextSegment segment = TextSegment.from("hello");
+        Embedding embedding = embeddingModel.embed(segment.text()).content();
+        contextRunner
+                .withPropertyValues(
+                        "langchain4j.redis.enabled=true",
```
Comment (langchain4j) [inline langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java:52]: I am wondering, should we ask users to _explicitly_ enable this?
In my understanding, if user adds `langchain4j-redis-spring-boot-starter` into dependencies, it is pretty obvious that he wants to use it, so we should automatically create a `RedisEmbeddingStore` with meaningful defaults, such as default host (localhost), port (6379), random index name, dimensions of an `EmbeddingModel` present in the context (we should probably add `EmbeddingModel.dimensions()` method to enable this.

File: langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                        "langchain4j.redis.host=" + redis.getHost(),
+                        "langchain4j.redis.port=" + redis.getFirstMappedPort(),
+                        "langchain4j.redis.indexName=" + randomUUID(),
+                        "langchain4j.redis.dimension=" + 384,
+                        "langchain4j.redis.metadataFieldsName=test-key"
```
Comment (langchain4j) [inline langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java:57]:
```suggestion
                        "langchain4j.redis.metadataKeys=test-key"
```

File: langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                )
+                .run(context -> {
+                    EmbeddingStore<TextSegment> embeddingStore = context.getBean(RedisEmbeddingStore.class);
```
Comment (langchain4j) [inline langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java:60]: I would also add a check that `context.getBean(EmbeddingStore.class)` returns `RedisEmbeddingStore`

File: pom.xml [hunk 6]
```diff
@@ -15,22 +15,27 @@
...
+        <module>langchain4j-redis-spring-boot-starter</module>
     </modules>
 
     <properties>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <spring.boot.version>2.7.18</spring.boot.version>
+        <tinylog.version>2.6.2</tinylog.version>
+        <langchain4j.embeddings.version>0.25.0</langchain4j.embeddings.version>
```
Comment (langchain4j) [inline pom.xml:27]:
```suggestion
        <langchain4j.embeddings.version>0.28.0</langchain4j.embeddings.version>
```

**Review by langchain4j — COMMENTED** [review]
@Martin7-1 thank you a lot and so sorry for the late review, somehow I overlooked PRs in this repository :(

File: langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java (hunk 5 above)
Comment (Martin7-1) [reply langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java:60]: The next line of code check it by `assertThat(embeddingStore).isInstanceOf(RedisEmbeddingStore.class)`.

File: langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java (hunk 3 above)
Comment (Martin7-1) [reply langchain4j-redis-spring-boot-starter/src/test/java/dev/langchain4j/store/embedding/redis/spring/RedisEmbeddingStoreAutoConfigurationIT.java:52]: hmmm... That's a good point. Will change it after adding `EmbeddingModel.dimensions()`

**Comment by Martin7-1** [comment]
@langchain4j Hi! I've been done it with adding default dimension by `EmbeddingModel.dimension()`.

**Comment by Martin7-1** [comment]
Note: Thit PR should be merged after #29

**Review by langchain4j — APPROVED** [review]
@Martin7-1 thank you!

**Pull request merged**